### ISSUE TITLE
Launching container fails when APPTAINER_BINDPATH is set

### ISSUE THREAD

**Issue by chrisburr** [apptainer/apptainer#1143]
### Version of Apptainer

```bash
$ /cvmfs/oasis.opensciencegrid.org/mis/apptainer/bin/apptainer --version
apptainer version 1.1.6-1.el7
```

### Expected behavior

Launching the container to succeed.

### Actual behavior

When running a container with the `APPTAINER_BINDPATH` environment variable set I see an error like `error: while mounting ... destination ... doesn't exist in container`. Using `--bind` instead works. I've managed to stumble upon a reproducer that seems to reliably reproduce the failure provided it's running from your home directory.

### Steps to reproduce this behavior

```bash
$ cd ~
$ mkdir -p a/b && touch a/d && cd a
$ APPTAINER_BINDPATH=$PWD /cvmfs/oasis.opensciencegrid.org/mis/apptainer/bin/apptainer exec --home $PWD/b --bind $PWD/d /cvmfs/cernvm-prod.cern.ch/cvm3 pwd
FATAL:   container creation failed: mount hook function failure: mount /afs/cern.ch/user/c/cburr/a/d->/afs/cern.ch/user/c/cburr/a/d error: while mounting /afs/cern.ch/user/c/cburr/a/d: destination /afs/cern.ch/user/c/cburr/a/d doesn't exist in container
```

* Running from `/tmp` instead of `$HOME` fixes the issue
* Replacing `APPTAINER_BINDPATH=$PWD` with `--bind $PWD` fixes the issue

### What OS/distro are you running

I've seen this failure on both CentOS 7 and Arch Linux.

### How did you install Apptainer

Seen with multiple but using `/cvmfs/oasis.opensciencegrid.org/mis/apptainer/bin/apptainer` to make it more easily reproducible.

**Comment by GodloveD** [comment]
I'm able to reproduce this with the following:
```
cd ~
mkdir -p a/b && touch a/d && cd a
APPTAINER_BINDPATH=$PWD apptainer exec --home $PWD/b --bind $PWD/d docker://alpine pwd
```
And I will also note that I do _not_ see this issue with v1.1.5.

**Comment by DrDaveD** [comment]
> And I will also note that I do _not_ see this issue with v1.1.5.

I disagree with that.  All of the apptainer versions are installed under /cvmfs/oasis.opensciencegrid.org/mis/apptainer, and if I select any of them I get the same error.  Likewise with singularity (by replacing "apptainer" with "singularity" in the above path and changing APPTAINER_BINDPATH to SINGULARITY_BINDPATH) back through singularity 3.6.  Only back on singularity 3.5 does the problem go away.

This is related to #742, which is scheduled to be addressed in v1.2.0.  However it shows a slightly different behavior.  I will mark this to be addressed by @cclerget along with that.

> - Running from /tmp instead of $HOME fixes the issue
> - Replacing APPTAINER_BINDPATH=$PWD with --bind $PWD fixes the issue

Another way to work around it is to not set APPTAINER_BINDPATH at all.

**Comment by GodloveD** [comment]
Oh you are right @DrDaveD I also do see the error with v1.1.5.  I must have messed up the reproducer somehow on my first attempt.
